src/store/modules/util/actions.ts

```diff
@@ -0,0 +1,32 @@
+import { ActionTree } from 'vuex'
+import RootState from '@/store/RootState'
+import UtilState from './UtilState'
+import * as types from './mutation-types'
+import { UtilService } from '@/services/UtilService'
+import { hasError } from '@/utils'
+
+const actions: ActionTree<UtilState, RootState> = {
+  async fetchRejectReasons({ commit }) {
+    try {
+      const payload = {
+        "inputFields": {
+          "parentEnumTypeId": "REPORT_AN_ISSUE"
+        },
+        "fieldList": ["enumCode", "description"],
+        "distinct": "Y",
+        "entityName": "EnumTypeChildAndEnum",
+        "viewSize": 20 // keeping view size 20 as considering that we will have max 20 reasons
+      }
+
+      const resp = await UtilService.fetchRejectReasons(payload)
+
+      if(!hasError(resp) && resp.data.count > 0) {
```

Comment:
I think we should clear last set value. Initialise with empty array, update if found from API and commit. This way we could avoid using the values of 2 different OMS in case of failure